### PR TITLE
Update TallGrass

### PR DESCRIPTION
**Pull request by DuoIncure**
## Introduction
Added functionality for Bonemeal to work on regular and fern versions of TallGrass.
Added ID checks for other possible blocks to be placed on.

## Tests
[Video](https://streamable.com/mxscbm)

### CODE REVIEW

File: src/block/TallGrass.php [hunk 1]
```diff
@@ -69,4 +70,23 @@ public function getFlameEncouragement() : int{
 	public function getFlammability() : int{
 		return 100;
 	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if($item instanceof Fertilizer){
+			$item->pop();
+			$blockAbove = $this->getPosition()->getWorld()->getBlockAt((int)$this->position->x, (int)$this->position->y + 1, (int)$this->position->z);
```
Comment (JavierLeon9966) [inline src/block/TallGrass.php]:
```suggestion
			$blockAbove = $this->position->getWorld()->getBlock($this->position->up());
```

File: src/block/TallGrass.php [hunk 2]
```diff
@@ -39,15 +40,15 @@ public function canBeReplaced() : bool{
 
 	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
 		$down = $this->getSide(Facing::DOWN)->getId();
-		if($down === BlockLegacyIds::GRASS or $down === BlockLegacyIds::DIRT){
+		if($down === BlockLegacyIds::GRASS || $down === BlockLegacyIds::DIRT || $down === BlockLegacyIds::FARMLAND || $down === BlockLegacyIds::PODZOL || $down === BlockLegacyIds::MYCELIUM){
```
Comment (JavierLeon9966) [inline src/block/TallGrass.php:43]: This change is unrelated

File: src/block/TallGrass.php [hunk 3]
```diff
@@ -39,15 +40,15 @@ public function canBeReplaced() : bool{
...
 			return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
 		}
 
 		return false;
 	}
 
 	public function onNearbyBlockChange() : void{
-		if($this->getSide(Facing::DOWN)->isTransparent()){ //Replace with common break method
+		if($this->getSide(Facing::DOWN)->isTransparent()){
```
Comment (JavierLeon9966) [inline src/block/TallGrass.php:51]: This change is unrelated

File: src/block/TallGrass.php [hunk 4]
```diff
@@ -69,4 +70,23 @@ public function getFlameEncouragement() : int{
...
+			$item->pop();
+			$blockAbove = $this->getPosition()->getWorld()->getBlockAt((int)$this->position->x, (int)$this->position->y + 1, (int)$this->position->z);
+
+			if($blockAbove->getId() === BlockLegacyIds::AIR){
+				$doubleBlock = match ($this->getMeta()) {
+					BlockLegacyMetadata::TALLGRASS_NORMAL => VanillaBlocks::DOUBLE_TALLGRASS(),
+					BlockLegacyMetadata::TALLGRASS_FERN => VanillaBlocks::LARGE_FERN(),
+				};
+				$this->position->getWorld()->setBlock($this->position, $doubleBlock, true);
+				$this->position->getWorld()->setBlock($this->getSide(Facing::UP)->getPosition(), $doubleBlock->setTop(true), true);
```
Comment (JavierLeon9966) [inline src/block/TallGrass.php]:
```suggestion
				$this->position->getWorld()->setBlock($this->position->up(), $doubleBlock->setTop(true), true);
```

**Review by JavierLeon9966 — CHANGES_REQUESTED** [review]

**Comment by dktapps** [comment]
This PR should be split up. Please don't submit multiple unrelated changes in one PR.

**Comment by dktapps** [comment]
Closing due to combination of changes. Please resubmit the changes in separate PRs.

**Pull request closed without merging**